### ISSUE TITLE
Wrap guide doesn't correctly update after changing font size or family in settings view

### ISSUE THREAD

**Issue by izuzak** [atom/wrap-guide#25]
This is a continuation from https://github.com/atom/atom/issues/3254#issuecomment-74852872 (cc @mdekauwe).

When the font size is increased and decreased with commands, the wrap guide updates correctly. Here's a video:

![font](https://cloud.githubusercontent.com/assets/38924/6245949/f35bd818-b763-11e4-8487-551fd83dab1d.gif)

However, if the font size is modified via the settings view, it doesn't update correct (it does update, but not to the expected position). Here's a video:

![font](https://cloud.githubusercontent.com/assets/38924/6247264/a87fa8c4-b772-11e4-9072-d78d66a1f234.gif)

Notice that the font size is 11. Then I modify it to 14 in settings view and go back to the editor. The wrap guide moved only slightly to the right (not as much as expected). After that I _decrease_ the font size with `cmd -` to 13, but notice that the wrap guide moves to the _right_ instead of to the left which is what should happen when you decrease font size. This happened because the wrap guide wasn't at the correct position before I hit `cmd -`, and then moved to the correct position after I hit `cmd -`.

cc @benogle

**Comment by thomasjo** [comment]
This is no longer an issue—I'm unable to reproduce this in v0.33.0.

**Comment by mdekauwe** [comment]
This is still an issue for me and I have version 1.0.2, so I don't think this was fixed!?

**Comment by benogle** [comment]
This works for me on editors that are visible, is broken on invisible editors.

* open some editors
* open settings
* change font size
* Tab over to an editor
* Wrap guide is not in the right place.

cc @as-cii This could be a regression?

**Comment by as-cii** [comment]
Probably, but I’d have to see a reference to a commit or a PR that fixed this problem to give an accurate answer. Hopefully, I'll be able to tell by writing a failing spec and bisecting to a moment in the (Atom | wrap-guide) git history where things worked correctly. :white_check_mark: 

Thanks @benogle for providing additional reproduction steps, I’ll investigate this real soon. :mag_right:

**Comment by as-cii** [comment]
I was unable to confirm this was caused by a regression (admittedly because I haven’t spent too much time digging), but the following is what I think it’s happening.

When initializing wrap guides we attach the following callback, so that we can get notified as soon as the font size configuration changes:

```coffee
subscriptions.add atom.config.onDidChange 'editor.fontSize', ->
  # setTimeout because we need to wait for the editor measurement to happen
  setTimeout(updateGuideCallback, 0)
```

This is done for all the editors, visible and hidden ones, although two different things happen on each kind of editor:

1. Visible editors get notified of the font size change and immediately trigger a font sampling on `TextEditorComponent`. As a result, `defaultCharWidth` is computed right away and it’s available as soon as the `setTimeout` above gets called.
2. Hidden editors defer measurements until it becomes visible. This means that, although `atom.config.onDidChange 'editor.fontSize'` is triggered as soon as the user changes the configuration parameter, the actual character width gets calculated when the user actually focuses on the hidden editor.

In other words, as of now, `wrap-guide` is relying on an incorrect character width value for hidden editors.   Thus, I’d say this was not a regression, as the component is supposed to work exactly this way and the fact that it worked in the past may have been accidental.

A possible solution could be to have a dedicated event on `TextEditor`, which fires as soon as `setDefaultCharWidth` gets called (e.g. `onDidChangeDefaultCharacterWidth`); aside from solving this problem, that would allow us to get rid of the `setTimeout` call as well. :fire: 

@benogle: thoughts? :thought_balloon:

**Comment by technicalpickles** [comment]
This is still an issue in 1.0.19. I originally filed https://github.com/atom/atom/issues/9125 because I didn't quite get a search leading me here, sooooooo including Preferred Line Length for SEO :grin:

**Comment by lee-dohm** [comment]
Updated title based on #55.

**Comment by greencopper** [comment]
This is still an issue on Linux using version 1.8.0.

**Comment by greencopper** [comment]
I just updated to BETA and the problem is gone. Thanks.

**Comment by pcperini** [comment]
Any updates on this? Still seeing it on Atom v1.9.8 (OS X 10.11.6)

**Comment by YoYoYonnY** [comment]
Still in Atom 1.11.1 (See atom/atom#13009)

**Comment by lukescott** [comment]
Having this issue on Atom 1.17.2 on OS X 10.12. Using commands works correctly, but when you change it via settings it does not update.

**Comment by emesterhazy** [comment]
This is still an issue when changing the font family via settings on Atom 1.19.3 on Windows 7. Restarting Atom resets the guide and serves as a good workaround luckily.

**Comment by 50Wliu** [comment]
Fixed by using the very helpful `getNextUpdatePromise` method.  Should be out starting with Atom 1.24.
